lib/azure/core/http/http_request.rb

```diff
@@ -149,7 +154,7 @@ def call
 
           response = HttpResponse.new(res)
           response.uri = uri
-          raise response.error unless response.success?
+          raise response.error if !response.success? && !@has_retry_filter
```

Comment:
@yaxia if we add the '!@hasretry_filter' to the condition, would we ever raise an error when there's a retry filter? are we ignoring the response success/failure if there's a retry filter?